### PR TITLE
Replace usage of transforms with update transforms

### PR DESCRIPTION
**Pull request by thebrianchen**
Porting from [web](https://github.com/firebase/firebase-js-sdk/pull/4153) and [Android](https://github.com/firebase/firebase-android-sdk/pull/2262).

First commit contains initial port changes. Second commit changes contains code removal for `localTransformResults`, which was done in Android and backported to [web](https://github.com/firebase/firebase-js-sdk/pull/4225).

### CODE REVIEW

**Review by schmidt-sebastian — CHANGES_REQUESTED** [review]
I am surprised by how pretty this looks. Thanks for doing this!

**Review by thebrianchen — COMMENTED** [review]
thanks!

File: Firestore/core/src/local/local_serializer.cc [hunk 1]
```diff
@@ -328,9 +329,38 @@ MutationBatch LocalSerializer::DecodeMutationBatch(
...
+      HARD_ASSERT(
+          proto.writes[i].which_operation ==
+              google_firestore_v1_Write_update_tag,
+          "TransformMutation should be preceded by a patch or set mutation");
+      _google_firestore_v1_Write new_mutation{current_mutation};
+      new_mutation.update_transforms_count =
+          transform_mutation.transform.field_transforms_count;
+      new_mutation.update_transforms =
+          MakeArray<_google_firestore_v1_DocumentTransform_FieldTransform>(
+              transform_mutation.transform.field_transforms_count);
```
Comment (schmidt-sebastian) [inline Firestore/core/src/local/local_serializer.cc:354]: Do we still need this call?

**Review by schmidt-sebastian — APPROVED** [review]

File: Firestore/core/src/local/local_serializer.cc (hunk 1 above)
Comment (thebrianchen) [reply Firestore/core/src/local/local_serializer.cc]: removed.

**Pull request merged**